File: src/main/java/fun/spmc/smpmod/discord/config/DiscordConfig.java
package fun.spmc.smpmod.discord.config;

public record DiscordConfig(String webhook, String token, String messageChannelId) {
    public DiscordConfig {
        if (webhook == null) webhook = "";
        if (token == null) token = "";
        if (messageChannelId == null) messageChannelId = "";
    }

    public DiscordConfig() {
        this("", "", "");
    }
}

File: src/main/java/spmc/smpmod/discord/config/DiscordConfig.kt
package spmc.smpmod.discord.config

class DiscordConfig @JvmOverloads constructor(webhook: String? = null, token: String? = null, messageChannelId: String? = null) {
	val webhook: String = webhook?: ""
	val token: String = token?: ""
	val messageChannelId = messageChannelId?: ""
}